### PR TITLE
Feature: stub Redis

### PR DESCRIPTION
**Pull request by doncem**
### Change description ###

Ability to enable/disable Redis access via feature flag.

```bash
./bin/run-in-docker.sh
...
...
no record of Redis
...
...

./bin/run-in-docker.sh -p REDIS_ENABLED=true
...
...
2018-01-15T21:49:55.299+0000 INFO  [main] org.redisson.Version:41: Redisson 3.5.7
2018-01-15T21:49:57.908+0000 WARN  [main] Unable to connect to Redis server: localhost/127.0.0.1:6379
```

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by doncem** [comment]
Investigating swagger docs publisher and first step - bypass redisson client

File: src/main/java/uk/gov/hmcts/reform/sendletter/cache/SentLettersCache.java [hunk 1]
```diff
@@ -12,21 +10,29 @@
...
         this.ttlInSeconds = ttlInSeconds;
+        this.isEnabled = true;
+    }
+
+    public SentLettersCache() {
+        this.redisCache = null;
+        this.checksumChecker = null;
+        this.ttlInSeconds = 0;
+        this.isEnabled = false;
     }
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/cache/SentLettersCache.java]: wouldn't it be clearer to just have another implementation?

File: src/test/java/uk/gov/hmcts/reform/sendletter/cache/SentLettersStubbedCacheTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.sendletter.cache;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.sendletter.model.Letter;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/sendletter/cache/SentLettersStubbedCacheTest.java]: mockito is not used

File: src/test/java/uk/gov/hmcts/reform/sendletter/cache/SentLettersStubbedCacheTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        // then
+        assertThat(added).isTrue();
+    }
+
+    @Test
+    public void should_never_call_real_code_and_void() {
+        // when
+        cache.remove(new Letter());
+
+        // then
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/sendletter/cache/SentLettersStubbedCacheTest.java]: something missing here

File: src/main/java/uk/gov/hmcts/reform/sendletter/config/RedisConfiguration.java [hunk 4]
```diff
@@ -1,26 +1,40 @@
...
+            Redisson.create(config),
+            new LetterChecksumGenerator(),
+            ttlInSeconds
+        );
+    }
+
+    @Bean
+    @ConditionalOnProperty(name = "redis.enabled", havingValue = "false")
+    public SentLettersCache sentLettersDisabledCache() {
+        return new SentLettersCache();
```
Comment (KarolJastrzebski) [inline src/main/java/uk/gov/hmcts/reform/sendletter/config/RedisConfiguration.java]: I am not sure if this is a good sign when flags begin to creep in and modify behaviour. What about an alternative approach with interface and two implementations? Both `SentLettersRedisCache` and `SentLettersNullCache` would implement `SentLettersCache`. Actual decision about cache behaviour is made upon instantiation of the object and not on runtime. Not to mention cleaner and simpler code in both implementations.

**Pull request closed without merging**